interview-questions.md

```diff
@@ -7,90 +7,97 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 3. What is a higher-order function?
 
-  Your answer:
+  Your answer: A higher order function are functions that operate on other functions. They eitehr take other functions are arguments or return them.
 
-  Researched answer:
+  Researched answer: A higher order function is a function that accepts and/or returns another function. It is considered "higher-order" because instead of dealing with strings, numbers, or booleans, it takes it one step further to operate on functions. They can be used to store functions as variables, use them in arrays, assign them as methods, pass arguemnts, or return them from other functions. 
 
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: JSX is a combination of HTML and Javascript fully named JavaScript XML. It lets us write HTML in React.
 
-  Researched answer:
+  Researched answer: JSX stands for Javascript XML. It allows us to write HTML as a JavaScript syntax extension in React. This way it can be passed to the DOM without any "createElement()" and/or "appendChild()". It is not required but it makes it easier to write React apps.  
 
 
 
 5. What is React?
 
-  Your answer:
+  Your answer: React is a Javascript libray developed by Facebook. It utilizes Node Modules and components to generate very fast, dynamic, applications that only refresh the component that needs changing through the DOM and virtual DOM.
+
 
-  Researched answer:
+  Researched answer: React is a Javascript Libray developed by Facebook. It makes creating interactive UIs much more painless. It is used for "client side" programming" which makes it a Frontend library. It utilizes the DOM and virtual DOM to make updates to single components on a page or app as seamless and efficient as possible.
 
 
 
 6. What is yarn? What file(s) are updated when you run `$ yarn`?
 
-  Your answer:
+  Your answer: Yarn is a Javascript packages manager. When $ yarn is ran it updates/generates your node-modules.
 
-  Researched answer:
+  Researched answer: Yarn is a new JavaScript package manager that aims for speed, and reliability. It allows you to use and share code with other developers. It is shared through a package. More specifically package.json that has all the code being shared. All the packages installed are often referred to as "modules". It was also developed by the almighty Facebook. Running '$ yarn' will default to 'yarn install" this will install the dependencies defined in a package.json file. 
 
 
 
 7. What is localhost:3000? How do we use localhost:3000 in the development process?
 
-  Your answer:
+  Your answer: Localhost:3000 is the local port used to display and run our React app on our local host server (our computer only). We use it so we can view our React app all compiled in our web browser. It will also display any errors if it runs into any while being compiled. 
```

Comment:
Totally, but its not exactly responsible for showing errors, thats up to the manager/application itself if it supports displaying live errors.